### DISCUSSION TITLE
Error

### DISCUSSION THREAD

**Discussion by rschwend** [adorobis/hacomfoairmqtt#57] [General]
Hi

On HassOS.

config.ini:
`
[DEFAULT]
# Serial port CA350 RS232 / RS485 direct or via USB TTL adapter
SerialPort=/dev/ttyUSB0
# Select RS232 or RS485 protocol
RS485_protocol=False
# Interval in seconds at which data from RS232 will be polled
refresh_interval=10
# automatically enable PC Mode (disable comfosense)
enablePcMode=False
debug=True

[DEVICE]
FanOutAbsent=15
FanOutLow=25
FanOutMid=40
FanOutHigh=70
FanInAbsent=15
FanInLow=25
FanInMid=40
FanInHigh=70
SetUpFanLevelsAtStart=True

[MQTT]
# MQTT broker - IP
MQTTServer=192.168.3.14
# MQTT broker - Port
MQTTPort=1883
 # MQTT broker - keepalive
MQTTKeepalive=45
# MQTT broker - user - default: empty (disabled/no authentication)
MQTTUser=mqtt
# MQTT broker - password - default: empty (disabled/no authentication)
MQTTPassword=NOT-MyPassword
`

I get the following error:

➜  ca350 python3 /config/custom_components/ca350/ca350runner.py &
[1] 7439
I am parent process:                                                                                                                                              
Process ID: 7439
Child's process ID: 7443

I am child process:
Process ID: 7443
Parent's process ID: 7439
Run ca350.py
[1]  + 7439 done       python3 /config/custom_components/ca350/ca350runner.py
➜  ca350 
➜  ca350 *****************************
* CA350 MQTT Home Assistant *
*****************************

Traceback (most recent call last):
  File "/config/custom_components/ca350/ca350.py", line 1136, in <module>
    mqttc = mqtt.Client('CA350')
            ^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/ca350/python3venv/lib/python3.11/site-packages/paho/mqtt/client.py", line 766, in __init__
    raise ValueError(
ValueError: Unsupported callback API version: version 2.0 added a callback_api_version, see migrations.md for details
Run ca350.py

**Comment by adorobis** [comment]
Looks like a breaking change in the new pahoo mqtt client. I'd recommend to use older pahoo client version for now until I check how to make this backward compatible and update the code for the new version. 
Which deployment are you using? Your own environment or the docker?

**Reply by adorobis** [reply]
Anyway, you need to downgrade the pahoo-mqtt package to the version 1, e.g. the latest is 1.6.1. Depending on how you have installed it you should install it with `pip install pahoo-mqtt==1.6.1` or `apt install python3-pahoo-mqtt==1.6.1`

**Comment by adorobis** [comment]
I've updated the Docker file and the installation instructions. Won't migrate to the version 2 of mqtt client for now.

**Comment by adorobis** [comment]
Edit: just noticed you are on HassOS. So if you have followed the installation instructions you have to do the following:
```
python3 -m venv /config/custom_components/ca350/python3venv
source /config/custom_components/ca350/python3venv/bin/activate
pip3 uninstall paho-mqtt
pip3 install paho-mqtt==1.6.1
deactivate

**Comment by rschwend** [comment]
This workaround is working perfect! Thank you!
